cmd/promtool/main.go

```diff
@@ -1151,6 +1154,61 @@ func (j *jsonPrinter) printLabelValues(v model.LabelValues) {
 	json.NewEncoder(os.Stdout).Encode(v)
 }
 
+type unittestPrinter struct {
+	Step model.Duration
+}
+
+func (u *unittestPrinter) printValue(v model.Value) {
+	switch modelType := v.(type) {
+	case model.Vector:
+		samples := make(map[string][]string)
+
+		for _, samplePair := range modelType {
+			metricName := samplePair.Metric.String()
+
+			if _, ok := samples[metricName]; !ok {
+				samples[metricName] = []string{}
+			}
+
+			samples[metricName] = append(samples[metricName], strconv.FormatFloat(float64(samplePair.Value), 'f', -1, 64))
+		}
+
+		inputSeries := make([]series, 0, len(samples))
+
+		for metric, value := range samples {
+			inputSeries = append(inputSeries, series{Series: metric, Values: strings.Join(value, " ")})
+		}
+
+		yaml.NewEncoder(os.Stdout).Encode(testGroup{Interval: u.Step, InputSeries: inputSeries})
```

Comment:
Probably worth going up another level with this to `unitTestFile` for easier redirection